onnx/cpp2py_export.cc

```diff
@@ -674,72 +687,90 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
     result.SerializeToString(&out);
-    return py::bytes(out);
+    return nb::bytes(out.c_str(), out.size());
   });
 
   // Submodule `inliner`
   auto inliner = onnx_cpp2py_export.def_submodule("inliner");
   inliner.doc() = "Inliner submodule";
 
-  inliner.def("inline_local_functions", [](const py::bytes& bytes, bool convert_version) {
+  inliner.def("inline_local_functions", [](const nb::bytes& bytes, bool convert_version) {
     ModelProto model{};
     ParseProtoFromPyBytes(&model, bytes);
     inliner::InlineLocalFunctions(model, convert_version);
     std::string out;
     model.SerializeToString(&out);
-    return py::bytes(out);
+    return nb::bytes(out.c_str(), out.size());
   });
 
   // inline_selected_functions: Inlines all functions specified in function_ids, unless
   // exclude is true, in which case it inlines all functions except those specified in
   // function_ids.
   inliner.def(
       "inline_selected_functions",
-      [](const py::bytes& bytes, std::vector<std::pair<std::string, std::string>> function_ids, bool exclude) {
+      [](const nb::bytes& bytes, std::vector<std::pair<std::string, std::string>> function_ids, bool exclude) {
         ModelProto model{};
         ParseProtoFromPyBytes(&model, bytes);
         auto function_id_set = inliner::FunctionIdSet::Create(std::move(function_ids), exclude);
         inliner::InlineSelectedLocalFunctions(model, *function_id_set);
         std::string out;
         model.SerializeToString(&out);
-        return py::bytes(out);
+        return nb::bytes(out.c_str(), out.size());
       });
 
   inliner.def(
       "inline_selected_functions2",
-      [](const py::bytes& bytes, std::vector<std::pair<std::string, std::string>> function_ids, bool exclude) {
+      [](const nb::bytes& bytes, std::vector<std::pair<std::string, std::string>> function_ids, bool exclude) {
         ModelProto model{};
         ParseProtoFromPyBytes(&model, bytes);
         auto function_id_set = inliner::FunctionIdSet::Create(std::move(function_ids), exclude);
         inliner::InlineSelectedFunctions(model, *function_id_set, nullptr);
         std::string out;
         model.SerializeToString(&out);
-        return py::bytes(out);
+        return nb::bytes(out.c_str(), out.size());
       });
 
   // Submodule `shape_inference`
   auto shape_inference = onnx_cpp2py_export.def_submodule("shape_inference");
   shape_inference.doc() = "Shape Inference submodule";
-  py::register_exception<InferenceError>(shape_inference, "InferenceError");
+  nb::exception<InferenceError>(shape_inference, "InferenceError");
 
-  py::class_<InferenceContext> inference_context(shape_inference, "InferenceContext", "Inference context");
+  nb::class_<InferenceContext> inference_context(shape_inference, "InferenceContext", "Inference context");
 
-  inference_context.def("get_attribute", &InferenceContext::getAttribute);
+  inference_context.def("get_attribute", [](InferenceContext& self, size_t idx) -> nb::object {
```

Comment:
size_t idx should be replaced by const std::string& name or const char* name

Reply:
Fixed